aten/src/THNN/generic/MultiMarginCriterion.c

```diff
@@ -119,13 +154,14 @@ void THNN_(MultiMarginCriterion_updateGradInput)(
 	       "inconsistent target size");
   }
 
-  g = (sizeAverage ? 1./((real)(nframe*dim)) : 1./((real)dim));
+  g = (sizeAverage && reduce ? 1./((real)(nframe*dim)) : 1./((real)dim));
 
   input = THTensor_(newContiguous)(input);
   target = THIndexTensor_(newContiguous)(target);
   input_data = THTensor_(data)(input);
 
   THTensor_(resizeAs)(gradInput, input);
+  gradInput = THTensor_(newContiguous)(gradInput);
```

Comment:
Is this needed?

Reply:
I discussed this with @zou3519 earlier on another PR. I think we reached the conclusion that resizeAs gives contiguous tensor if the tensor was actually resized, so this call is needed if a resize didn't happen. However, this call could potentially be unnecessary if gradInput is always resized, or guaranteed to be contiguous if it is the correct size.

Reply:
Right. However, here `gradInput` is an output, so it is guaranteed to be contiguous.

Reply:
I'm not 100% sure it's guaranteed to be contiguous. Could we remove this newContiguous and assert that gradInput is contiguous?